src/httprpc.cpp

```diff
@@ -294,21 +296,26 @@ static bool InitRPCAuthentication()
 {
     if (gArgs.GetArg("-rpcpassword", "") == "")
     {
-        LogInfo("Using random cookie authentication.\n");
-
         std::optional<fs::perms> cookie_perms{std::nullopt};
         auto cookie_perms_arg{gArgs.GetArg("-rpccookieperms")};
         if (cookie_perms_arg) {
             auto perm_opt = InterpretPermString(*cookie_perms_arg);
             if (!perm_opt) {
-                LogInfo("Invalid -rpccookieperms=%s; must be one of 'owner', 'group', or 'all'.\n", *cookie_perms_arg);
+                LogError("Invalid -rpccookieperms=%s; must be one of 'owner', 'group', or 'all'.\n", *cookie_perms_arg);
                 return false;
             }
             cookie_perms = *perm_opt;
         }
 
-        if (!GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)) {
-            return false;
+        if (std::optional<bool> result{GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)}) {
+            if (*result) {
+                LogInfo("Using random cookie authentication.\n");
+            } else {
+                // Failure reason message expected to already have been logged.
+                return false;
+            }
+        } else {
+            LogInfo("RPC authentication cookie file generation is disabled.");
         }
     } else {
         LogPrintf("Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.\n");
```

Comment:
You mean
```
LogInfo("Using random cookie authentication.\n");
```
was run independently of whether `GenerateAuthCookie` returned `true` or `false`, but now it may not be? It seemed wrong to me so I made it only happen when the cookie was actually generated.

If you mean that
```
LogPrintf("Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.\n");
```
was emitted when `GenerateAuthCookie` was `true`, I think you may be mixing up some braces/indentation.

Reply:
Yes, I was confused by indentation, resolve the comment por favor